mava/environment_loop.py

```diff
@@ -550,22 +554,23 @@ def should_terminate(episode_count: int, step_count: int) -> bool:
                 num_steps is not None and step_count >= num_steps
             )
 
-        def should_run_loop(eval_condtion: Tuple) -> bool:
+        def should_run_loop(eval_condition: Tuple) -> bool:
             """Check if the eval loop should run in current step.
 
             Args:
                 eval_condition : tuple containing interval key and count.
 
             Returns:
-                a bool indicatings if eval should run.
+                a bool indicating if eval should run.
             """
             should_run_loop = False
             eval_interval_key, eval_interval_count = eval_condition
             counts = self.get_counts()
+
             if counts:
-                count = counts.get(eval_interval_key)
-                # We run eval loops around every eval_interval_count (not exactly every
-                # eval_interval_count due to latency in getting updated counts).
+                count = counts[eval_interval_key]
```

Comment:
This is not a problem for this PR but the eval interval specified and actual is quite different which is not ideal for papers
For example, when interval is 2000, 
``[evaluator/0] Running eval loop at executor step: 119350
[evaluator/0] Running eval loop at executor step: 122150
[evaluator/0] Running eval loop at executor step: 125050``

Reply:
The most common seems to be to evaluate after a set amount of executor steps for a set amount of episodes. But we have the flexibility to trigger based on any count parameter.